docs/utilities/parser.md

```diff
@@ -2,168 +2,85 @@
+???+ warning
+    This will increase the compressed package size by >10MB due to the Pydantic dependency.
 
-#### event_parser decorator
+    To reduce the impact on the package size at the expense of 30%-50% of its performance[Pydantic can also be
+    installed without binary files](https://pydantic-docs.helpmanual.io/install/#performance-vs-package-size-trade-off){target="_blank" rel="nofollow"}:
 
-Use the decorator for fail fast scenarios where you want your Lambda function to raise an exception in the event of a malformed payload.
+    Pip example:`SKIP_CYTHON=1 pip install --no-binary pydantic aws-lambda-powertools[parser]`
 
-`event_parser` decorator will throw a `ValidationError` if your event cannot be parsed according to the model.
 
-???+ note
-    **This decorator will replace the `event` object with the parsed model if successful**. This means you might be careful when nesting other decorators that expect `event` to be a `dict`.
+You can also add as a dependency in your preferred tool: e.g., requirements.txt, pyproject.toml.
 
-```python hl_lines="19" title="Parsing and validating upon invocation with event_parser decorator"
-from aws_lambda_powertools.utilities.parser import event_parser, BaseModel
-from aws_lambda_powertools.utilities.typing import LambdaContext
-from typing import List, Optional
+### Data Model with Parse
 
-import json
+Define models by inheriting from `BaseModel` to parse incoming events. Pydantic then validates the data, ensuring all fields adhere to specified types and guaranteeing data integrity.
 
-class OrderItem(BaseModel):
-	id: int
-	quantity: int
-	description: str
+#### Event parser
 
-class Order(BaseModel):
-	id: int
-	description: str
-	items: List[OrderItem] # nesting models are supported
-	optional_field: Optional[str] = None # this field may or may not be available when parsing
+The `event_parser` decorator automatically parses and validates the event.
 
 
-@event_parser(model=Order)
-def handler(event: Order, context: LambdaContext):
-	print(event.id)
-	print(event.description)
-	print(event.items)
+```python title="getting_started_with_parser.py"
+--8<-- "examples/parser/src/getting_started_with_parser.py"
+```
 
-	order_items = [item for item in event.items]
-	...
+The `@event_parser(model=MyEvent)` automatically parses the event into the specified Pydantic model MyEvent.
 
-payload = {
-	"id": 10876546789,
-	"description": "My order",
-	"items": [
-		{
-			"id": 1015938732,
-			"quantity": 1,
-			"description": "item xpto"
-		}
-	]
-}
+The function catches **ValidationError**, returning a 400 status code with an error message if the input doesn't match the `MyEvent` model. It provides robust error handling for invalid inputs.
 
-handler(event=payload, context=LambdaContext())
-handler(event=json.dumps(payload), context=LambdaContext()) # also works if event is a JSON string
-```
+#### Parse function
```

Comment:
Hey @anafalcao, there is something weird here. Pls take a look

![image](https://github.com/user-attachments/assets/84c2c7b1-1263-4612-b329-7655216481eb)

Reply:
Fixed it! It was breaking due to a "---" prior to the sub header